Dockerfile

```diff
@@ -1,18 +1,32 @@
 ## Global Args #################################################################
 ARG BASE_UBI_IMAGE_TAG=9.3-1476
+
+ARG FLASH_ATTN_VERSION=1.0.9
+ARG FLASH_ATTN_V2_VERSION=2.3.6
+
 ARG PROTOC_VERSION=25.1
+
 #ARG PYTORCH_INDEX="https://download.pytorch.org/whl"
 ARG PYTORCH_INDEX="https://download.pytorch.org/whl/nightly"
-ARG PYTORCH_VERSION=2.3.0.dev20231221
+#ARG PYTORCH_VERSION=2.3.0.dev20231221
+ARG PYTORCH_VERSION=2.2.0.dev20231213
```

Comment:
With the nightly PyTorch versions, I think there's a chance of incompatibilities unless the exact build of PyTorch matches with the dependencies. The prebuilt wheel for Flash Attention v2 was compiled with `2.2.0.dev20231127` [REF](https://github.com/Dao-AILab/flash-attention/blob/92dd5703ecdb99aa4a4aee9817f28557907403a2/.github/workflows/publish.yml#L47).

It also could just work... but the safest thing would be to exactly match the versions.

Reply:
Well, if we did want to keep the PyTorch nightly build tags in sync for the Flash Attention v2 build and this here TGIS build, we may need to looks at the tagged version of the build/publish workflow file, i.e.:

Flash | Torch | Ref |
---|---|---|
2.5.0 | 2.2.0.dev2023**1130** | [REF](https://github.com/Dao-AILab/flash-attention/blob/v2.5.0/.github/workflows/publish.yml#L47)
2.4.3 | 2.2.0.dev2023**1130** | [REF](https://github.com/Dao-AILab/flash-attention/blob/v2.4.3/.github/workflows/publish.yml#L47)
2.4.2 | 2.2.0.dev2023**1106** | [REF](https://github.com/Dao-AILab/flash-attention/blob/v2.4.2/.github/workflows/publish.yml#L47)
2.4.1 | 2.2.0.dev2023**1106** | [REF](https://github.com/Dao-AILab/flash-attention/blob/v2.4.1/.github/workflows/publish.yml#L47)
2.4.0 | 2.2.0.dev2023**1106** | [REF](https://github.com/Dao-AILab/flash-attention/blob/v2.4.0/.github/workflows/publish.yml#L47)
2.3.6 | 2.2.0.dev2023**1127** | [REF](https://github.com/Dao-AILab/flash-attention/blob/v2.3.6/.github/workflows/publish.yml#L47)
2.3.5 | 2.1.0 (**no dev**) | [REF](https://github.com/Dao-AILab/flash-attention/blob/v2.3.5/.github/workflows/publish.yml#L47)

Reply:
Yup, https://github.com/IBM/text-generation-inference/actions/runs/7706334748/job/21001719349?pr=12#step:6:696

```
 > [cpu-tests  2/11] RUN pip install torch=="2.2.0.dev20231127+cpu" --index-url "https://download.pytorch.org/whl/nightly/cpu" --no-cache-dir:
0.587 Looking in indexes: https://download.pytorch.org/whl/nightly/cpu
0.931 ERROR: Could not find a version that satisfies the requirement torch==2.2.0.dev20231127+cpu (from versions: 2.2.0.dev20231010+cpu, 2.2.0.dev20231205+cpu, 2.2.0.dev20231206+cpu, 2.2.0.dev20231207+cpu, 2.2.0.dev20231208+cpu, 2.2.0.dev20231209+cpu, 2.2.0.dev20231210+cpu, 2.2.0.dev20231211+cpu, 2.2.0.dev20231212+cpu, 2.2.0.dev20231213+cpu, 2.3.0.dev20231214+cpu, 2.3.0.dev20231215+cpu, 2.3.0.dev20231216+cpu, 2.3.0.dev20231217+cpu, 2.3.0.dev20231218+cpu, 2.3.0.dev20231219+cpu, 2.3.0.dev20231220+cpu, 2.3.0.dev20231221+cpu, 2.3.0.dev20231222+cpu, 2.3.0.dev20231223+cpu, 2.3.0.dev20231224+cpu, 2.3.0.dev20231225+cpu, 2.3.0.dev20231226+cpu, 2.3.0.dev20231227+cpu, 2.3.0.dev20231228+cpu, 2.3.0.dev20231229+cpu, 2.3.0.dev20231230+cpu, 2.3.0.dev20231231+cpu, 2.3.0.dev20240101+cpu, 2.3.0.dev20240102+cpu, 2.3.0.dev20240103+cpu, 2.3.0.dev20240104+cpu, 2.3.0.dev20240105+cpu, 2.3.0.dev20240106+cpu, 2.3.0.dev20240107+cpu, 2.3.0.dev20240108+cpu, 2.3.0.dev20240109+cpu, 2.3.0.dev20240110+cpu, 2.3.0.dev20240111+cpu, 2.3.0.dev20240113+cpu, 2.3.0.dev20240114+cpu, 2.3.0.dev20240115+cpu, 2.3.0.dev20240116+cpu, 2.3.0.dev20240117+cpu, 2.3.0.dev20240118+cpu, 2.3.0.dev20240119+cpu, 2.3.0.dev20240120+cpu, 2.3.0.dev20240121+cpu, 2.3.0.dev20240122+cpu, 2.3.0.dev20240123+cpu, 2.3.0.dev20240124+cpu, 2.3.0.dev20240125+cpu, 2.3.0.dev20240126+cpu, 2.3.0.dev20240127+cpu, 2.3.0.dev20240128+cpu, 2.3.0.dev20240129+cpu)
0.932 ERROR: No matching distribution found for torch==2.2.0.dev20231127+cpu
------
Dockerfile:181
```

Reply:
`torch==2.2.0` works